internal/cli/template_create.go

```diff
@@ -0,0 +1,184 @@
+package cli
+
+import (
+	"os"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateCreateCommand struct {
+	*baseCommand
+
+	flagSummary                    string
+	flagExpandedSummary            string
+	flagReadmeMarkdownTemplatePath string
+	flagWaypointHCLTemplatePath    string
+	flagTFCNoCodeModuleSource      string
+	flagTFCNoCodeModuleVersion     string
+	flagTags                       []string
+}
+
+func (c *ProjectTemplateCreateCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	if len(args) != 1 {
+		c.ui.Output("Single argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	name := args[0]
+
+	var template pb.ProjectTemplate
+
+	template.Name = name
+	if c.flagSummary != "" {
+		template.Summary = c.flagSummary
+	}
+	if c.flagExpandedSummary != "" {
+		template.ExpandedSummary = c.flagExpandedSummary
+	}
+
+	if c.flagReadmeMarkdownTemplatePath != "" {
+		rmt, err := os.ReadFile(c.flagReadmeMarkdownTemplatePath)
+		if err == os.ErrNotExist {
+			c.ui.Output("Readme template file does not exist: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		} else if err != nil {
+			c.ui.Output("Unable to read readme.md template file: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+		template.ReadmeMarkdownTemplate = rmt
+	}
+
+	if c.flagWaypointHCLTemplatePath != "" {
+		wpt, err := os.ReadFile(c.flagWaypointHCLTemplatePath)
```

Comment:
We check for `os.ErrNotExist` when reading the template path but we don't distinguish here